front/components/vaults/VaultDataSourceViewContentList.tsx

```diff
@@ -210,28 +210,40 @@ export const VaultDataSourceViewContentList = ({
     viewType: isValidContentNodesViewType(viewType) ? viewType : undefined,
   });
 
-  const { hasContent: hasDocuments } = useStaticDataSourceViewHasContent({
-    owner,
-    dataSourceView,
-    parentId,
-    viewType: "documents",
-  });
-  const { hasContent: hasTables } = useStaticDataSourceViewHasContent({
-    owner,
-    dataSourceView,
-    parentId,
-    viewType: "tables",
-  });
+  const { hasContent: hasDocuments, isLoading: isDocumentsLoading } =
+    useStaticDataSourceViewHasContent({
+      owner,
+      dataSourceView,
+      parentId,
+      viewType: "documents",
+    });
+  const { hasContent: hasTables, isLoading: isTablesLoading } =
+    useStaticDataSourceViewHasContent({
+      owner,
+      dataSourceView,
+      parentId,
+      viewType: "tables",
+    });
 
   useEffect(() => {
-    if (viewType === undefined) {
-      if (hasTables === true && hasDocuments === false) {
-        handleViewTypeChange("tables");
-      } else if (hasDocuments !== undefined) {
-        handleViewTypeChange("documents");
-      }
+    // If the view only has content in one of the two views, we switch to that view.
+    // if both view have content, or neither views have content, we default to documents.
+    if (hasTables === true && hasDocuments === false) {
```

Comment:
Shouldn't we check the loading state here as well, and wait for both to be loaded before taking actions?